lib/common/schemas.c

```diff
@@ -1147,77 +1149,77 @@ update_validation(xmlNode **xml_blob, int *best, int max, gboolean transform,
         }
         rc = pcmk_ok;
         local_best = lpc;
-        if (transform) {
-            xmlNode *upgrade = NULL;
-            pcmk__schema_t *next_schema = NULL;
-            next = lpc+1;
-
-            if (next > max_stable_schemas) {
-                /* There is no next version */
-                crm_trace("Stopping at %s", schema->name);
-                break;
-            }
 
-            if (max > 0 && (lpc == max || next > max)) {
-                crm_trace("Upgrade limit reached at %s (lpc=%d, next=%d, max=%d)",
-                          schema->name, lpc, next, max);
-                break;
-            }
+        if (!transform) {
+            /* Validation with this schema succeeded. We are not doing
+             * transforms, so try the next schema using the same XML.
+             */
+            lpc++;
+            continue;
+        }
```

Comment:
I find the style you've switched to here to almost always be much easier to follow.  I'm a big fan of this commit.